samples/java-application-with-reflection/pom.xml

```diff
@@ -136,12 +136,23 @@
                                 <phase>package</phase>
                             </execution>
                         </executions>
+                        <!-- tag::native-plugin-agent-options[] -->
                         <configuration>
+                            <!-- end::native-plugin-agent-options[] -->
                             <imageName>${imageName}</imageName>
                             <buildArgs>
                                 <buildArg>--no-fallback</buildArg>
                             </buildArgs>
+                            <!-- tag::native-plugin-agent-options[] -->
+                            <agentOptions name="exec">
```

Comment:
I think we miss an enabled attribute like `<agentOptions enabled=true name="exec">` here, similar to what we have on Gradle side.

Reply:
I don't think it makes sense. The `agent=true` flag in Gradle _delegates to_ `agentOptions.enabled=true`. So it's effectively the _same flag_. It's there to enable the agent. Then you have the options. The options are defined in the build file (build.gradle or pom.xml) per "profile", where the profile is explicit in Maven (`name="exec"`) and _implicit_ in Gradle (because you run the `main` native image, it will use the `main` options).

So what @sbrannen did looks good to me.

Reply:
In https://github.com/graalvm/native-build-tools/pull/168#issuecomment-978017722, I made the following point.

> having the "enable agent" flag standalone allows users to enable the agent within the build file even if they are not defining custom agent options.

We do not want to force users to declare an `<agentOptions>` block if they are not using custom agent options.

Reply:
This is in line with the Gradle support.

The Gradle DSL does not force users to declare agent options to enable the agent.

Reply:
What I care about is allowing to use agent options out of the box with Maven without custom command line parameters. It seems that will be achieved via applying unamed agent options (not yet done) so I will close this conversation since that will be handled differently.